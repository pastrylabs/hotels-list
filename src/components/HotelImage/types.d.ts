type HotelImageProps = {
  imageSrc: string;
  caption: string;
  offer?: string;
};
