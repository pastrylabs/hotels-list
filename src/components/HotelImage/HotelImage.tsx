import React from "react";
import { Theme, createStyles, makeStyles } from "@material-ui/core/styles";
import GridList from "@material-ui/core/GridList";
import GridListTile from "@material-ui/core/GridListTile";
import GridListTileBar from "@material-ui/core/GridListTileBar";
import ProgressiveImage from "react-progressive-image";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    gridList: {
      width: 500,
      height: 450
    }
  })
);

const dominantImageColor = "#86356B";
const placeholder = (
  <div
    style={{ backgroundColor: dominantImageColor, height: 300, width: 500 }}
  />
);

const HotelImage = (props: HotelImageProps) => {
  const classes = useStyles();
  const { imageSrc, caption, offer } = props;

  return (
    <GridList cellHeight={180} className={classes.gridList}>
      <GridListTile>
        <ProgressiveImage src={imageSrc} placeholder="">
          {(src: string, loading: boolean) => {
            return loading ? (
              placeholder
            ) : (
              <>
                <img src={src} alt={caption} />
                <noscript>
                  <img
                    className="progressive-image no-script"
                    src={src}
                    alt={caption}
                  />
                </noscript>
              </>
            );
          }}
        </ProgressiveImage>
        ;{offer ? <GridListTileBar title={offer} /> : null}
      </GridListTile>
    </GridList>
  );
};

export default HotelImage;
