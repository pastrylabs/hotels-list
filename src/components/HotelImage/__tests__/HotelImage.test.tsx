import React from "react";
import { render } from "@testing-library/react";
import HotelImage from "../HotelImage";

it("Render image", () => {
  const { container } = render(
    <HotelImage imageSrc="assets/img" caption="description" />
  );
  expect(container).toMatchSnapshot();
});

it("Render image with offer", () => {
  const { getByText, getByAltText } = render(
    <HotelImage
      imageSrc="assets/img"
      caption="description"
      offer="exclusive deal"
    />
  );
  expect(getByText(/exclusive deal/i)).toBeInTheDocument();
});
