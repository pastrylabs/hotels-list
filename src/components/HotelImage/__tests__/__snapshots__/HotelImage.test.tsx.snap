// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render image 1`] = `
<div>
  <ul
    class="MuiGridList-root makeStyles-gridList-1"
    style="margin: -2px;"
  >
    <li
      class="MuiGridListTile-root"
      style="width: 50%; height: 184px; padding: 2px;"
    >
      <div
        class="MuiGridListTile-tile"
      >
        <div
          style="background-color: rgb(134, 53, 107); height: 300px; width: 500px;"
        />
      </div>
    </li>
  </ul>
</div>
`;
